boards/PDM/Inc/Io/Io_Efuse.h

```diff
@@ -0,0 +1,145 @@
+ * @param fsb_port Handle to Efuse's fault status GPIO port
+ * @param fsb_pin Handle to Efuse's fault status GPIO pin
+ * @param channel0_port Handle to Efuse's channel 0 GPIO port
+ * @param channel0_pin Handle to Efuse's channel 0 GPIO pin
+ * @param channel1_port Handle to Efuse's channel 1 GPIO port
+ * @param channel1_pin Handle to Efuse's channel 1 GPIO pin
+ * @return The created Efuse, whose ownership is given to the caller
+ */
+struct Efuse_Context *Io_Efuse_Create(
+    SPI_HandleTypeDef *hspi,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin,
+    GPIO_TypeDef *     fsob_port,
+    uint16_t           fsob_pin,
+    GPIO_TypeDef *     fsb_port,
+    uint16_t           fsb_pin,
+    GPIO_TypeDef *     channel0_port,
+    uint16_t           channel0_pin,
+    GPIO_TypeDef *     channel1_port,
+    uint16_t           channel1_pin);
+
+/**
+ * Deallocate the memory used by the given Efuse
+ * @param e_fuse The Efuse to deallocate
+ */
+void Io_Efuse_Destroy(struct Efuse_Context *e_fuse);
+
+/**
+ * Enable the channel 0 output
+ */
+void Io_Efuse_Channel0Enable(const struct Efuse_Context *const e_fuse);
+
+/**
+ * Disable the channel 0 output
+ */
+void Io_Efuse_Channel0Disable(const struct Efuse_Context *const e_fuse);
+
+/**
+ * Enable the channel 1 output
+ */
+void Io_Efuse_Channel1Enable(const struct Efuse_Context *const e_fuse);
+
+/**
+ * Disable the channel 1 output
+ */
+void Io_Efuse_Channel1Disable(const struct Efuse_Context *const e_fuse);
+
+/**
+ * Set the current/temperature monitoring option of CSNS pin
+ * @param selection The desired monitoring function which can be selected by
+ * specifying the following values:
+ *                  CSNS_FUNCTION_DISABLED - Current sensing disabled
+ *                  CSNS_FUNCTION_CURRENT_CH0 - Current sensing for Channel 0
+ *                  CSNS_FUNCTION_CURRENT_CH1 - Current sensing for Channel 1
+ *                  CSNS_FUNCTION_TEMPERATURE - Efuse temperature sensing
+ *                  CSNS_FUNCTION_CURRENT_SUM - Current sensing for summed
+ * channels (for Parallel mode)
+ * @param e_fuse Pointer to the Efuse structure for the Efuse being configured
+ * @return EXIT_CODE_OK if the channel configuration was successful
+ *         EXIT_CODE_TIMEOUT if one of the SPI writes timed-out
+ */
+ExitCode Io_Efuse_ConfigureChannelMonitoring(
+    uint8_t                     selection,
+    struct Efuse_Context *const e_fuse);
+
+/**
+ * Exit fail-safe mode and disable the watchdog timer
+ * @param e_fuse Pointer to the Efuse structure for the Efuse being configured
+ * @return EXIT_CODE_OK if the efuse exited fail-safe mode
+ *         EXIT_CODE_TIMEOUT if the one of the SPI writes timed-out
+ *         EXIT_CODE_UNIMPLEMENTED if the efuse failed to exit fail-safe mode
+ */
+ExitCode Io_Efuse_ExitFailSafeMode(struct Efuse_Context *const e_fuse);
+
+/**
+ * Check if the efuse's channel 0 or channel 1 enters fault mode
```

Comment:
For sentences that have "Check if the efuse's ...." I'd change it to "Check if the **_given_** efuses's .... " since this file will be shared with multiple efuses.

Reply:
changed